src/pineko/evolve.py

```diff
@@ -309,32 +376,61 @@ def evolve_grid(
         )
         for mur2 in mur2_grid
     ]
-    # We need to use ekompatibility in order to pass a dictionary to pineappl
-    fktable = grid.evolve(
-        ekompatibility.pineappl_layout(operators),
-        xir * xir * mur2_grid,
-        alphas_values,
-        "evol",
-        order_mask=order_mask,
-        xi=(xir, xif),
-    )
+
+    def prepare(operator, items):
+        (q2, _), op = items
+        info = PyOperatorSliceInfo(
+            fac0=operator.mu20,
+            x0=operator.bases.inputgrid.raw,
+            pids0=basis_rotation.evol_basis_pids,
+            fac1=q2,
+            x1=operator.bases.targetgrid.raw,
+            pids1=operator.bases.targetpids,
+            pid_basis=PyPidBasis.Evol,
+        )
+        return (info, op.operator)
+
+    if operators2 is not None:
+        fktable = grid.evolve_with_slice_iter2(
+            map(lambda it: prepare(operators1, it), operators1.items()),
+            map(lambda it: prepare(operators2, it), operators2.items()),
+            ren1=mur2_grid,
+            alphas=alphas_values,
+            xi=(xir, xif),
+            order_mask=order_mask,
+        )
+    else:
```

Comment:
as discussed: check that `operators1` matches what the grid wants (`polarized`/`time_like`) - similar above; if it does not match, just cry and leave it to the user to swap